middleware/proxy/upstream.go

```diff
@@ -159,10 +159,16 @@ func parseBlock(c *parse.Dispenser, u *staticUpstream) error {
 		if !c.Args(&header, &value) {
 			return c.ArgErr()
 		}
-		proxyHeaders.Add(header, value)
+		if u.proxyHeaders == nil {
+			u.proxyHeaders = make(http.Header)
+		}
+		u.proxyHeaders.Add(header, value)
 	case "websocket":
-		proxyHeaders.Add("Connection", "{>Connection}")
-		proxyHeaders.Add("Upgrade", "{>Upgrade}")
+		if u.proxyHeaders == nil {
+			u.proxyHeaders = make(http.Header)
+		}
```

Comment:
@mholt Yes, it will be more readable when we create this map once `staticUpstream` is created. I don't know whether we need add extra http header when proxying in most cases. If this is the rare  case, I think we be lazy like this to avoid unnecessary gc...